### DISCUSSION TITLE
keycloak openID connect with moodle

### DISCUSSION THREAD

**Discussion by pkchandana** [keycloak/keycloak#22242] [Q&A]
I am trying to integrate moodle with keycloak using openID connect but getting error "The URL is blocked" after keycloak login. I am unable to understand what can be the issue.

14:27
Here i am attaching the error..i tried to print the logs in oidcclient.php

12:45
Error in OpenID Connect. Please check logs for more information.

More information about this error

Debug info:
Error code: erroroidccall×Dismiss this notification
Stack trace:
line 49 of /auth/oidc/classes/utils.php: moodle_exception thrown
line 349 of /auth/oidc/classes/oidcclient.php: call to auth_oidc\utils::process_json_response()
line 306 of /auth/oidc/classes/loginflow/authcode.php: call to auth_oidc\oidcclient->tokenrequest()
line 136 of /auth/oidc/classes/loginflow/authcode.php: call to auth_oidc\loginflow\authcode->handleauthresponse()
line 168 of /auth/oidc/auth.php: call to auth_oidc\loginflow\authcode->handleredirect()
line 31 of /auth/oidc/index.php: call to auth_plugin_oidc->handleredirect()
×Dismiss this notification
Output buffer: http://test.com:8080/auth/realms/realm2022/protocol/openid-connect/tokenArray ( [client_id] => keycloak_prod [grant_type] => authorization_code [code] => cab18929-7a43-433e-aad2-14b95ddc4672.d87fa1df-b904-42bc-ab12-ec51fa5023ae.5f8bf63a-5832-4b0d-ab5a-6cda3c673c19 [redirect_uri] => http://www.example.com/auth/oidc/ [client_secret] => 8121c915-9524-441f-9db2-ef49f80b62fc ) The URL is blocked

**Comment by darius-m** [comment]
What is the redirect URL that you have configured? According to [the Moodle OAuth2 documentation](https://docs.moodle.org/402/en/OAuth_2_services) you should redirect the response  back to Moodle's `/admin/oauth2callback.php` endpoint.

**Reply by pkchandana** [reply]
![image](https://github.com/keycloak/keycloak/assets/81303758/3c6020f9-2f82-4f7f-9c70-5c97d18db114)
Yes /admin/oauth2callback.php is the redirect URL

**Reply by darius-m** [reply]
Is there any error in the Keycloak logs? Can you share what client flows are enabled in Keycloak and how the client is configured in Moodle?
